src/Microsoft.AspNet.Security/Infrastructure/AuthenticationHandler.cs

```diff
@@ -307,7 +311,18 @@ protected virtual void ApplyResponseCore()
         protected virtual async Task ApplyResponseCoreAsync()
         {
             await ApplyResponseGrantAsync();
-            await ApplyResponseChallengeAsync();
+
+            // If authenticate was called and the the status is still 401, authZ failed so set 403 and stop
+            // REVIEW: Does this need to ensure that there's the 401 is challenge for this auth type?
```

Comment:
Actually, per our discussion, this should move back into ApplyResponseChallenge because CookieAuth is going to redirect to an AccessDenied page instead.